src/elfpy/simulators.py

```diff
@@ -280,11 +280,26 @@ def setup_simulated_entities(self, override_dict=None):
             self.agents = {}
         self.market.log_market_step_string()
         # continue adding other users
-        for policy_number, policy_name in enumerate(self.config.simulator.user_policies):
+        for policy_number, policy_instruction in enumerate(self.config.simulator.user_policies):
+            if ":" in policy_instruction:  # we have custom parameters
```

Comment:
This looks a lot cleaner. I think it would be a good idea to also add a check function that receives the custom parameter and verifies the format with asserts. This way we provide more useful error messaging in situations where the custom parameters are formatted incorrectly or trying to do something that the user can't.

Reply:
added in validate_custom_parameters() but needs testing, left in as a TODO

Reply:
added testing, seems to work